### ISSUE TITLE
Update version of Terraform because of GPG issue

### ISSUE THREAD

**Issue by denizgenc** [alphagov/cyber-security-concourse-base-image#46]
Running `terraform init` with version 0.12.26 emits the following message:

```
registry.terraform.io: This version of Terraform has an outdated GPG key and is unable to verify new provider releases. Please upgrade Terraform to at least 0.12.31 to receive new provider updates. For details see: https://discuss.hashicorp.com/t/hcsec-2021-12-codecov-security-event-and-hashicorp-gpg-key-exposure/23512
```

`Dockerfile` and `amazonlinux2/Dockerfile` should be upgraded as per the message

**Comment by danjoneslf** [comment]
This should be partially solved by #48 - enabling TF env. We still need version files / tf install/use to switch version.

**Comment by danjoneslf** [comment]
The default TF version for the container is now 0.12.31 so this is done.
